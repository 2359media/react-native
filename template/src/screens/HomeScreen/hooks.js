import {useState, useCallback, useEffect} from 'react';
import {useSelector, useDispatch} from 'react-redux';
import {useNavigation} from '@react-navigation/native';
import {useRequest} from '../../utils/apiUtils';
import TodoApi from '../../apis/TodoApi';
import TodoAction from '../../actions/TodoAction';
import AccountApi from '../../apis/AccountApi';
import AppAction from '../../actions/AppAction';

export const useTodosState = () => {
  const dispatch = useDispatch();
  const request = useRequest();

  const [{loading, refreshing, error}, setState] = useState({});
  const todos = useSelector(s => s.todo.all);

  const reload = useCallback(
    (refresh = true) => {
      setState({loading: true, refreshing: refresh, error: null});
      request(TodoApi.getAll())
        .then(r => {
          dispatch(TodoAction.set(r.data));
          setState({loading: false, refreshing: false});
        })
        .catch(e => {
          setState({loading: false, refreshing: false, error: e.statusText});
        });
    },
    [dispatch, request],
  );

  useEffect(() => reload(false), [reload]);

  return {loading, refreshing, error, reload, todos};
};

export const useLogout = () => {
  const dispatch = useDispatch();
  const request = useRequest();
  const navigation = useNavigation();

  function logout() {
    request(AccountApi.logout(), {showLoading: true}).finally(() => {
      navigation.reset({index: 0, routes: [{name: 'Login'}]});

      // make sure transition is finished to avoid glitch, could be better
      setTimeout(() => dispatch(AppAction.setToken(null)), 1000);
    });
  }

  return logout;
};
