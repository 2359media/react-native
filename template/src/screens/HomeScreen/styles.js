import {createStylesheet} from '../../utils/styleUtils';
import Color from '../../constants/Color';

const styles = createStylesheet({
  div: {
    marginBottom: 16,
  },
  item: safeArea => ({
    paddingHorizontal: safeArea.left + 24,
    paddingVertical: 16,
    borderBottomColor: '#EEE',
    borderBottomWidth: 1,
    backgroundColor: Color.WHITE,
  }),
  itemText: {
    fontSize: 16,
    fontWeight: 'bold',
  },
});

export default styles;
