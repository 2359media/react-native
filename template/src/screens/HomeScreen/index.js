import React from 'react';
import {RefreshControl, View, Text, TouchableHighlight} from 'react-native';
import {useSafeArea} from 'react-native-safe-area-context';
import Screen from '../../components/Screen';
import ContentView from '../../components/ContentView';
import TopNavigation from '../../components/TopNavigation';
import Button from '../../components/Button';
import EmptyView from '../../components/EmptyView';
import {useTodosState, useLogout} from './hooks';
import styles from './styles';

const HomeScreen = ({navigation}) => {
  const safeArea = useSafeArea();
  const {loading, refreshing, error, todos, reload} = useTodosState();
  const logout = useLogout();

  function gotoNextScreen(id) {
    navigation.navigate('Details', {id});
  }

  return (
    <Screen>
      <TopNavigation
        title="Home"
        rightButtons={
          <Button
            testID="logoutButton"
            type="barItem"
            icon="logout"
            onPress={logout}
          />
        }
      />
      {todos.length > 0 ? (
        <ContentView
          testID="contentView"
          data={todos}
          renderItem={({item: {id, title}}) => (
            <TouchableHighlight
              testID={`item::${id}`}
              underlayColor="#999"
              onPress={() => gotoNextScreen(id)}>
              <View style={styles.item(safeArea)}>
                <Text style={styles.itemText}>{title}</Text>
              </View>
            </TouchableHighlight>
          )}
          refreshControl={
            <RefreshControl
              testID="refreshControl"
              refreshing={refreshing}
              onRefresh={reload}
            />
          }
        />
      ) : (
        <EmptyView
          testID="emptyView"
          loading={loading}
          onReload={reload}
          text={
            error ||
            (loading && 'Loading...') ||
            "You don't have anything to do"
          }
        />
      )}
    </Screen>
  );
};

export default HomeScreen;
