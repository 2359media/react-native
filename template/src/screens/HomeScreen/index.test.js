import React from 'react';
import {
  renderer,
  TestContainer,
  navigation,
  mockResponse,
} from '../../utils/testUtils';
import HomeScreen from '.';

const tree = renderer(<TestContainer Component={HomeScreen} />);

test('snapshot before fetching data', () => {
  expect(tree).toMatchSnapshot();
});

test('snapshot after fetching data', () => {
  tree.run();
  expect(tree).toMatchSnapshot();
});

test('goto details', () => {
  tree.run('item::1');
  expect(navigation.navigate).toBeCalledWith('Details', {id: 1});
});

test('empty data', () => {
  mockResponse(() => ({data: []}));

  tree.run('refreshControl', 'onRefresh');
  expect(tree.get('refreshControl')).toBeUndefined();
  expect(tree.get('emptyView', 'text')).toEqual(
    "You don't have anything to do",
  );
});

test('error screen', () => {
  mockResponse(() => ({status: 400, data: {message: 'User not registered'}}));

  tree.run('emptyView', 'onReload');
  expect(tree.get('emptyView', 'text')).toEqual('User not registered');
});

test('server error', () => {
  mockResponse(() => ({status: 503}));

  tree.run('emptyView', 'onReload');
  expect(tree.get('emptyView', 'text')).toEqual(
    'Request failed with status code 503',
  );
});

test('network error', () => {
  mockResponse(() => undefined);

  tree.run('emptyView', 'onReload');
  expect(tree.get('emptyView', 'text')).toEqual('Network Error');
});

test('logout', () => {
  //back to default routes
  mockResponse(null);

  tree.run('logoutButton');
  expect(navigation.reset).toBeCalledWith({
    index: 0,
    routes: [{name: 'Login'}],
  });
});
