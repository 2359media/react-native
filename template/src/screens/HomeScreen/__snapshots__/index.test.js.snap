// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot after fetching data 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "elevation": 5,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "shadowColor": "black",
          "shadowOpacity": 0.2,
          "shadowRadius": 5,
          "zIndex": 100,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 60,
          "paddingHorizontal": 16,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "minWidth": 50,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 8,
              "left": 8,
              "right": 8,
              "top": 8,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "margin": 4,
                "tintColor": "#EC632B",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "stretch",
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 8,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "black",
              "fontSize": 20,
              "fontWeight": "bold",
              "left": 50,
              "position": "absolute",
              "right": 50,
              "textAlign": "center",
            }
          }
        >
          Home
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "minWidth": 50,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 8,
              "left": 8,
              "right": 8,
              "top": 8,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID="logoutButton"
        >
          <Image
            source={1}
            style={
              Object {
                "margin": 4,
                "tintColor": "#EC632B",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Array [
        Object {
          "flexGrow": 1,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
        undefined,
      ]
    }
    data={
      Array [
        Object {
          "body": "provident occaecati excepturi optio reprehenderit quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
          "id": 1,
          "title": "sunt aut facere repellat ",
          "userId": 1,
        },
        Object {
          "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
          "id": 2,
          "title": "qui est esse",
          "userId": 1,
        },
        Object {
          "body": " repellat qui ipsa sit aut et iusto sed quo iure
voluptatem occaecati omnis eligendi aut ad
voluptatem doloribus vel accusantium quis pariatur
molestiae porro eius odio et labore et velit aut",
          "id": 3,
          "title": "ea molestias quasi exercitationem",
          "userId": 1,
        },
        Object {
          "body": "ullam et saepe reiciendis voluptatem adipisci
sit amet autem assumenda provident rerum culpa
quis hic commodi nesciunt rem tenetur doloremque ipsam iure
quis sunt voluptatem rerum illo velit",
          "id": 4,
          "title": "eum et est occaecati",
          "userId": 1,
        },
        Object {
          "body": "repudiandae veniam quaerat sunt sed
alias aut fugiat sit autem sed est
voluptatem omnis possimus esse voluptatibus quis
est aut tenetur dolor neque",
          "id": 5,
          "title": "nesciunt quas odio",
          "userId": 1,
        },
        Object {
          "body": "aperiam quia ut aspernatur corporis harum nihil quis provident sequi
mollitia nobis aliquid molestiae
perspiciatis et ea nemo ab reprehenderit accusantium quas
voluptate dolores velit et doloremque molestiae",
          "id": 6,
          "title": "dolorem eum magni eos",
          "userId": 1,
        },
        Object {
          "body": "dolore placeat quibusdam ea quo vitae
magni quis enim qui quis quo nemo aut saepe
quidem repellat excepturi ut quia
sunt ut sequi eos ea sed quas",
          "id": 7,
          "title": "magnam facilis autem",
          "userId": 1,
        },
        Object {
          "body": "dignissimos aperiam dolorem qui eum
facilis quibusdam animi sint suscipit qui sint possimus cum
quaerat magni maiores excepturi
ipsam ut commodi dolor voluptatum modi aut vitae",
          "id": 8,
          "title": "dolorem dolore est ipsam",
          "userId": 1,
        },
        Object {
          "body": "tempora et accusantium consectetur animi nesciunt iure dolore
enim quia ad
veniam autem ut quam aut nobis
et est aut quod aut provident voluptas autem voluptas",
          "id": 9,
          "title": "nesciunt iure omnis dolorem",
          "userId": 1,
        },
        Object {
          "body": "quo et expedita modi cum officia vel magni
doloribus qui repudiandae
vero nisi sit
quos veniam quod sed accusamus veritatis error",
          "id": 10,
          "title": "optio molestias id quia eum",
          "userId": 1,
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    keyboardShouldPersistTaps="handled"
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
        testID="refreshControl"
      />
    }
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="contentView"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="item::1"
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderBottomColor": "#EEE",
                "borderBottomWidth": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              sunt aut facere repellat 
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="item::2"
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderBottomColor": "#EEE",
                "borderBottomWidth": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              qui est esse
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="item::3"
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderBottomColor": "#EEE",
                "borderBottomWidth": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              ea molestias quasi exercitationem
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="item::4"
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderBottomColor": "#EEE",
                "borderBottomWidth": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              eum et est occaecati
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="item::5"
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderBottomColor": "#EEE",
                "borderBottomWidth": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              nesciunt quas odio
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="item::6"
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderBottomColor": "#EEE",
                "borderBottomWidth": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              dolorem eum magni eos
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="item::7"
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderBottomColor": "#EEE",
                "borderBottomWidth": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              magnam facilis autem
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="item::8"
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderBottomColor": "#EEE",
                "borderBottomWidth": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              dolorem dolore est ipsam
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="item::9"
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderBottomColor": "#EEE",
                "borderBottomWidth": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              nesciunt iure omnis dolorem
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="item::10"
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderBottomColor": "#EEE",
                "borderBottomWidth": 1,
                "paddingHorizontal": 24,
                "paddingVertical": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              optio molestias id quia eum
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`snapshot before fetching data 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "elevation": 5,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "shadowColor": "black",
          "shadowOpacity": 0.2,
          "shadowRadius": 5,
          "zIndex": 100,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 60,
          "paddingHorizontal": 16,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "minWidth": 50,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 8,
              "left": 8,
              "right": 8,
              "top": 8,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "margin": 4,
                "tintColor": "#EC632B",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "stretch",
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 8,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "black",
              "fontSize": 20,
              "fontWeight": "bold",
              "left": 50,
              "position": "absolute",
              "right": 50,
              "textAlign": "center",
            }
          }
        >
          Home
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "minWidth": 50,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 8,
              "left": 8,
              "right": 8,
              "top": 8,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID="logoutButton"
        >
          <Image
            source={1}
            style={
              Object {
                "margin": 4,
                "tintColor": "#EC632B",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#EEE",
        "flexGrow": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#999",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginTop": 16,
          "textAlign": "center",
        }
      }
    >
      You don't have anything to do
    </Text>
    <View
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 8,
          "left": 8,
          "right": 8,
          "top": 8,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#EC632B",
          "borderRadius": 8,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 16,
          "opacity": 1,
          "padding": 8,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "margin": 4,
          }
        }
      >
        Reload
      </Text>
    </View>
  </View>
</View>
`;
